pkg/ds/daemon_set_test.go

```diff
@@ -309,7 +307,7 @@ func TestSchedule(t *testing.T) {
 	labeled = labeledPods(t, ds)
 	Assert(t).AreEqual(len(labeled), expectedNodes, "Expected no nodes to be unlabeled")
 
-	err = waitForPodsInIntent(consulStore, len(beforeDeletePods))
+	err = waitForPodsInIntent(consulStore, 12)
```

Comment:
Then again, it's not very useful to wait for a number that's already there, since it will just exit immediately.